feat(plugin-sui): Add movepump action

ChainRex opened the pull request:
<!-- Use this template by filling in information and copying and pasting relevant items out of the HTML comments. -->

# Relates to

<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section must be filled out before the final review and merge. -->

# Risks

<!--
Low, medium, large. List what kind of risks and what could be affected.
-->
Medium:
- Financial transactions involving user funds
- Integration with external MovePump contract
- Complex multi-step transaction process

# Background

## What does this PR do?

This PR implements token creation for MovePump on Sui Network. The implementation follows a three-step process:

- Update coin template bytecode with custom token metadata
- Publish new coin package to Sui network
- Create token on MovePump

## What kind of change is this?
Features (non-breaking change which adds functionality)
<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there are no linked issues explaining why. If there is a related issue, it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If documentation change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?

1. Review the main implementation in src/actions/movepump.ts
2. Check bytecode update logic in src/utils.ts
3. Review token template in src/template/template_coin/

## Detailed testing steps
### 1. (Optional) Verify Template Bytecode
1. Compile the template_coin contract:
```
cd template/template_coin
sui move build
```
2. Generate bytecode and compare with constant:
```
xxd -c 0 -p build/template_coin/bytecode_modules/template_coin.mv | head -n 1
```
3. Verify the output matches `TEMPLATE_COIN_BYTECODE` in src/template/bytecode.ts
### 2. Test Token Creation via Bot Chat
1. Start conversation with Bot using this example format:
```
Create a new token called MyToken with symbol MTK on MovePump and buy 1 MTK. Here's the token info:
Description: My first token on MovePump
Image: https://example.com/token.png
Website: https://mytoken.com
Twitter: https://twitter.com/mytoken
Telegram: https://t.me/mytoken
```
2. Bot will process the request and return a response like:
```
Token MyToken (MTK) created successfully on MovePump!
Module ID: 0x[module_id]
Transaction: [transaction_hash]
View at: https://movepump.com/token/0x[module_id]::mtk::MTK
```
3. Visit the returned MovePump URL

0xnogo committed:
feat: Add new translations and update README structure

- Introduced new language translations for Eliza in Arabic, Chinese, German, Spanish, French, Hebrew, Hungarian, Italian, Japanese, Korean, Dutch, Polish, Portuguese, Romanian, Russian, Serbian, Tagalog, Thai, Turkish, and Vietnamese.
- Updated the README structure to include links to the new translations.
- Added a new plugin '@elizaos/plugin-hyperliquid' in the pnpm-lock.yaml file with its dependencies.
- Adjusted the README links to point to the correct paths for translations.

cygaar committed:
Add AGW support to the Abstract plugin

cygaar committed:
update lockfile

ai16z-demirix committed:
feature: adding tests for whatsapp plugin

mcp97 committed:
Merge pull request #2213 from ai16z-demirix/test/plugin-whatsapp

feature: adding tests for whatsapp plugin

cygaar committed:
Improve prompt

daizhengxue committed:
chore: sync pnpm-lock.yaml with develop branch

daizhengxue committed:
chore: sync pnpm-lock.yaml with develop branch

tcm390 committed:
Merge branch 'develop' into feature/add-deepseek-support

tcm390 committed:
Merge pull request #2186 from daizhengxue/feature/add-deepseek-support

fix: DeepSeek API bug:  missing api key setting

cygaar committed:
Improve prompt

ai16z-demirix committed:
improvement: using strict types to avoid erorrs like issue 2164

proteanx committed:
dont force root perm install

ai16z-demirix committed:
Merge branch 'develop' into fix/issue-2164

proteanx committed:
Merge branch 'develop' into noRootInstall

shakkernerd committed:
Merge pull request #2221 from proteanx/noRootInstall

fix: don't force root for install

shakkernerd committed:
Merge branch 'develop' into fix/issue-2164

HowieDuhzit committed:
Added Setup.sh to scripts

HowieDuhzit committed:
Change permissions of setup.sh to make it executable

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

sin-bufan committed:
fix: improve error handling and configuration checks in Farcaster client

- Updated safeParseInt to use Number.isNaN for better clarity.
- Added nullish coalescing to handle potential undefined FARCASTER_FID.
- Implemented a check to skip interactions if no FID is found.
- Enhanced stopping mechanism for the Farcaster client to ensure it only attempts to stop if the client is initialized.

gene-zhan committed:
feat: update pnpm-lock

HowieDuhzit committed:
Enhance setup script with improved error handling, logging, and character management features. Added support for verbose mode, character selection, and environment configuration. Updated dependency installation process and streamlined project setup. Removed obsolete code related to previous character management.

Commit:
Merge remote-tracking branch 'upstream/develop' into ai16zPR/tee-verifiable-log-api-from-develop

# Conflicts:
#	agent/src/index.ts
#	packages/core/src/types.ts
#	pnpm-lock.yaml

HowieDuhzit committed:
Refactor setup.sh to enhance logging functionality with fallback support for environments without gum. Ensure gum installation is prioritized before other operations. Clean up and streamline logging methods for better user experience.

HowieDuhzit committed:
Enhance NVM installation process in setup.sh by downloading directly from the official source and adding necessary environment variables to .bashrc and .profile. Ensure NVM directory is created if it doesn't exist, and improve loading checks for nvm.sh. This update streamlines the setup process for users.

HowieDuhzit committed:
Refactor NVM installation in setup.sh to streamline the process. Directly download and install NVM, source necessary scripts immediately, and improve error handling. Removed redundant checks and logging for installation failures, enhancing user experience during setup.

HowieDuhzit committed:
Refactor NVM installation in setup.sh to improve error handling and streamline the process. The installation script is now downloaded to a temporary file, made executable, and executed with enhanced logging for failures. Clean up of temporary files is also included to ensure a tidy environment.

HowieDuhzit committed:
Refactor NVM installation in setup.sh to streamline the process by directly executing the installation script without temporary files. This change simplifies the setup procedure and enhances user experience by reducing complexity and improving efficiency.

HowieDuhzit committed:
Refactor NVM installation in setup.sh to improve error handling and streamline the process. The installation now logs output to a temporary file, attempts to load NVM immediately after installation, and provides clearer error messages if the installation fails. This enhances user experience by ensuring better feedback during setup.

HowieDuhzit committed:
Remove deprecated NVM installation script and integrate NVM setup directly into setup.sh. The new implementation enhances error handling, logging, and user feedback during the Node.js and pnpm installation process. This streamlines the setup experience by ensuring NVM is installed and configured correctly before proceeding with other installations.

HashWarlock committed:
Merge pull request #1369 from artela-network/ai16zPR/tee-verifiable-log-api-from-develop

feat: RP for plugin-tee-verifiable-log

HowieDuhzit committed:
Refactor NVM installation in setup.sh to enhance error handling and streamline the setup process. The installation now directly downloads NVM from the official source, improves logging, and ensures necessary environment variables are set. This update simplifies user experience by providing clearer feedback and reducing complexity during setup.

HowieDuhzit committed:
Remove legacy setup scripts for macOS and Windows, consolidating functionality into a single start.sh script. This update enhances the installation process by integrating NVM, Node.js, and dependency management, while improving logging and error handling. The new script supports both macOS and Linux environments, streamlining user experience and reducing complexity in the setup process.

HowieDuhzit committed:
Refactor Chocolatey installation in start.ps1 script to improve logging and user feedback. The installation process is now more streamlined, with clearer messages during the setup of dependencies. This update enhances the overall user experience by ensuring that Chocolatey is installed correctly and efficiently before proceeding with other installations.

HowieDuhzit committed:
Refactor logging functions in start.ps1 script to improve clarity and consistency. Updated error, success, info, and verbose messages to use new custom functions, enhancing user feedback during the Chocolatey installation process.

HowieDuhzit committed:
Refactor logging and error handling in start.ps1 script to enhance user feedback and streamline installation processes. Updated logging functions to use custom messages for better clarity during dependency installation, NVM setup, and character selection. Improved error handling for various operations, ensuring clearer communication of issues and successful actions. This update enhances the overall user experience during the setup of the Eliza application.

HowieDuhzit committed:
Enhance logging functions in start.ps1 script by implementing CmdletBinding and mandatory parameters for improved clarity and consistency. This update ensures that error, success, info, and verbose messages provide better user feedback during the setup process, contributing to a more streamlined installation experience.

HowieDuhzit committed:
Refactor start.ps1 script to streamline setup process and enhance user feedback. Updated logging functions to improve clarity and consistency, ensuring better communication during the installation of WSL, Ubuntu, and Git. Integrated checks for WSL and Ubuntu installation, and added functionality to clone the Eliza repository if not present. This update simplifies the setup experience and provides clearer instructions for users.

HowieDuhzit committed:
Refactor start.ps1 script to simplify WSL and Ubuntu installation process. Streamlined logging functions for improved clarity and consistency in user feedback. Enhanced checks for WSL, Ubuntu, and Git installations, and ensured proper cloning of the Eliza repository. This update provides clearer instructions and a more efficient setup experience for users.

HowieDuhzit committed:
Refactor start.ps1 script to replace custom logging functions with direct Write-Host calls for improved clarity. Enhanced user feedback during WSL, Ubuntu, and Git installations, and streamlined the cloning process for the Eliza repository. This update simplifies the setup experience and provides clearer instructions for users.

HowieDuhzit committed:
Remove start.ps1 script and enhance start.sh for improved compatibility and user experience. The start.sh script now includes POSIX-compliant syntax, better error handling, and streamlined character selection logic. This update simplifies the setup process for the Eliza application by consolidating functionality and providing clearer instructions for users.

cygaar committed:
Fix agw tx sending

cygaar committed:
Unused types

HowieDuhzit committed:
Update README.md to enhance start script documentation and clarify automated setup process for Eliza. Added detailed instructions on using the start script, including character management and troubleshooting, to improve user experience.

Stumble committed:
feat: add support for gravity chain in EVM plugin

- Updated `fromChain` and `toChain` types in `index.ts` to include "gravity".
- Added "gravity" to the `EvmPluginConfig` interface in `index.ts`.

This enhancement allows the EVM plugin to handle transactions involving the "gravity" chain,
expanding the range of supported blockchain networks.

HowieDuhzit committed:
Update README.md and add Start Script Guide for improved user documentation. The README now links to a newly created guide detailing the start script's usage, system requirements, features, and troubleshooting steps, enhancing the overall user experience for setting up Eliza.

HowieDuhzit committed:
Update pnpm-lock.yaml to reflect dependency version upgrades and additions. Notable changes include updating 'utf-8-validate' to version 6.0.5 across multiple packages, adding 'langdetect' and 'jest' dependencies, and upgrading various packages such as 'zod' to version 3.24.1 and 'typescript' to 5.7.3. This commit enhances compatibility and ensures the project uses the latest stable versions of its dependencies.

HowieDuhzit committed:
Add NVM and Node.js setup functions to start.sh for improved environment configuration

This commit introduces two new functions in the start.sh script: install_nvm and setup_node. The install_nvm function automates the installation of NVM (Node Version Manager) and the specified Node.js version, while the setup_node function verifies the Node.js installation and installs pnpm if not already present. Enhanced logging provides better feedback during the setup process, ensuring a smoother experience for users configuring their Node.js environment.

HowieDuhzit committed:
Update pnpm-lock.yaml

HowieDuhzit committed:
Update pnpm-lock.yaml to modify the specifier for '@0glabs/0g-ts-sdk' from '0.2.1' to '0.2.1hjfgjfjdftyhjdh', ensuring accurate dependency tracking.

HowieDuhzit committed:
Refactor pnpm-lock.yaml to streamline dependencies and enhance project configuration. This update consolidates the lockfile structure, introduces a new package.json format, and ensures accurate dependency tracking. Key changes include the addition of essential scripts for build, start, and testing processes, as well as the integration of workspace management for improved organization. This commit enhances compatibility and prepares the project for future development.

HowieDuhzit committed:
Delete pnpm-lock.yaml

shakkernerd committed:
Merge branch 'develop' into pr/HowieDuhzit/2229

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2229 from HowieDuhzit/main

Merge my Eliza Installer with the current start.sh script

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

HowieDuhzit committed:
Add character creation template function to start.sh

This commit introduces a new function, create_character_template, in the start.sh script. This function generates a JSON template for character creation, including fields for name, clients, model provider, settings, bio, lore, knowledge, message examples, post examples, topics, style, and adjectives. The select_character function has been updated to utilize this new template function, streamlining the character creation process and enhancing user experience when managing characters in the Eliza application.

HowieDuhzit committed:
Merge branch 'develop' into develop

chuasonglin1995 committed:
fix readme and package.json

chuasonglin1995 committed:
Merge branch 'develop' into refactor-websearch

Lukapetro committed:
chore: resolve merge conflicts with develop

Lukapetro committed:
fix: env variable naming

KacperKoza343 committed:
fix: resolve Windows path issue in pnpm build client

Fixes #2223

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 9" notes

swizzmagik committed:
Merge branch 'develop' into feat/add-birdeye-plugin

MonteCrypto999 committed:
Add a service example

augchan42 committed:
git commit -m "feat(echochambers): add dead room detection and conversation starter

- Add multiple room support
- Implement proactive conversation starter for inactive rooms
- Add configurable quiet period and check intervals
- Add debug logging for conversation starter process

Closes #2246"

wtfsayo committed:
Merge pull request #2228 from Stumble/develop

feat: add support for gravity chain in EVM plugin

wtfsayo committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
Merge pull request #2243 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #9" notes

wtfsayo committed:
Merge branch 'develop' into agw_integration

Hack666r committed:
Minor typo in CHANGELOG.md

shakkernerd committed:
chore: remove multiple langdetect package

Commit:
Typo fix README.md

shakkernerd committed:
chore: console -> elizaLogger

cygaar committed:
ts ignore

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2207 from cygaar/agw_integration

feat: Add AGW support to the Abstract plugin

oxlupo committed:
Add Persian README file

oxlupo committed:
Add Persian README file

odilitime committed:
Merge pull request #2260 from oxlupo/farsi-readme

docs: Add Farsi (Persian )readme

0xCardiE committed:
add price per address functionality

0xCardiE committed:
Merge branch 'develop' into feature/gecko_price_per_addy

mcp97 committed:
Merge pull request #2262 from 0xCardinalError/feature/gecko_price_per_addy

feat: CoinGecko - add price per address functionality

augchan42 committed:
git commit -m "fix(db): add limit param to memory retrieval across adapters

- Add limit parameter to getMemoriesByRoomIds in SQLite adapter
- Add limit parameter to getMemoriesByRoomIds in SQLjs adapter
- Add limit parameter to getMemoriesByRoomIds in PGLite adapter
- Add limit parameter to getMemoriesByRoomIds in Supabase adapter
- Fix query parameter ordering in SQLjs adapter
- Add consistent DESC ordering across all adapters

Closes #2253"

Commit:
add POST /agents and DELETE /agents/:agentId

odilitime committed:
fix: test/lint develop (#2266)

* bump version

* add eslint config

* remove unused error var per lint

* fix lint

* comment out duplicate symbol

* prefix address as it's not used per lint

* add _ prefix for unused vars per lint

* remove unused error per lint

* prefix unused vars with _ per lint

* remove unused error per lint

* don't import unused ServiceType per lint

* remove unused vars per lint

* rename test per lint

* lint auto fixes

* deTS it per lint

* deTS per lint

* bump

Commit:
feat: Adding plugin for B² Network (#2010)

* feat: b2-network plugin

* docs:update readme and package description

* fix: solve build error

* modify

* modify

* feat: transfer action

* modify README.md

* docs: update readme

* chore: update env file

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* fix: default empty value and resolve type error

---------

Co-authored-by: goalongway <goalongway@163.com>
Co-authored-by: goalongway <36787789+goalongway@users.noreply.github.com>
Co-authored-by: goalongway <ma_huaibo@163.com>

shakkernerd committed:
Merge branch 'develop' into bugfix/optimize-recent-interactions-query

shakkernerd committed:
Merge pull request #2264 from augchan42/bugfix/optimize-recent-interactions-query

fix: (db) add limit param to memory retrieval across adapters

asyncjason committed:
Merge branch 'develop' into direct-client_agent-crud

odilitime committed:
Merge pull request #2256 from VitalikBerashvili/fix/typo

Typo fix README.md

odilitime committed:
Merge pull request #2255 from Hack666r/fix/typo

Minor typo in CHANGELOG.md

shakkernerd committed:
fix: add missing @elizaos/plugin-b2 to package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2268 from elizaOS/fix/plugin-b2

fix: missing @elizaos/plugin-b2

odilitime committed:
Merge pull request #2232 from HowieDuhzit/develop

Add character creation template function to start.sh

odilitime committed:
Merge branch 'develop' into feat/add-birdeye-plugin

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge branch 'develop' into koswald/voyageai

odilitime committed:
Merge pull request #2240 from blockydevs/2223--fix-pnpm-build-client-path

fix: resolve Windows path issue in pnpm build client

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge branch 'develop' into feature/squid-bridge

odilitime committed:
Merge branch 'develop' into koswald/voyageai

odilitime committed:
Merge pull request #1482 from Archethect/feature/squid-bridge

feat: Add cross chain swaps through Squid Router

odilitime committed:
Merge pull request #1710 from thetechnocratic/plugin-evm-oz-governance

feat: Plugin evm oz governance

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge pull request #1442 from Firbydude/koswald/voyageai

feat: Add support for VoyageAI embeddings API

odilitime committed:
Merge pull request #1417 from swizzmagik/feat/add-birdeye-plugin

feat: add birdeye plugin

wtfsayo committed:
Merge branch 'develop' into refactor-farcaster-client

sin-bufan committed:
refactor: simplify memory structure in FarcasterInteractionManager

- Removed the hash property from the memory content object.
- Updated the way memory is passed to the runtime processActions method to include the cast property in the content object.

asyncjason committed:
Merge branch 'develop' into direct-client_agent-crud

Commit:
don't need create, set works for this. catch startAgent errors. add PREVENT_UNHANDLED_EXIT support

Commit:
Merge branch 'direct-client_agent-crud' of https://github.com/jseparovic/eliza into direct-client_agent-crud

Commit:
fix merge issues. add back comments

lincheoll committed:
Fix: IME causes multiple messages on Enter (Fixes #2272)

lincheoll committed:
Merge branch 'develop' into 2272--fix-ime-multiple-messages

chuasonglin1995 committed:
Merge remote-tracking branch 'upstream/develop' into refactor-websearch

v1xingyue committed:
support load character from  character_url

v1xingyue committed:
Merge branch 'develop' into character-from-url

YoungPhlo committed:
docs: Add weekly contributor meeting notes

anonymouscommitter committed:
feat: add getMemoryByIds to database adapters

Lukapetro committed:
chore: resolve merge conflicts with develop

shakkernerd committed:
Revert "feat: Add support for VoyageAI embeddings API"

shakkernerd committed:
Merge pull request #2290 from elizaOS/revert-1442-koswald/voyageai

Revert "feat: Add support for VoyageAI embeddings API"

wtfsayo committed:
Merge branch 'develop' into docs/contributors-meetings

wtfsayo committed:
Merge pull request #2285 from YoungPhlo/docs/contributors-meetings

docs: Add weekly contributor meeting notes

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: export b2Plugin

shakkernerd committed:
Merge branch 'develop' into fixes

shakkernerd committed:
Merge pull request #2291 from elizaOS/fixes

Fixes

wtfsayo committed:
Merge branch 'develop' into direct-client_agent-crud

v1xingyue committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Merge pull request #2249 from PIPELINE-BTC/feat-add-a-service-example

feat: Add an example service

0xCardiE committed:
add all code

0xCardiE committed:
Merge branch 'develop' into feat/gitcoin_passport_plugin

v1xingyue committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Merge pull request #2296 from 0xCardinalError/feat/gitcoin_passport_plugin

feat: Gitcoin passport

tcm390 committed:
Merge pull request #2274 from lincheoll/2272--fix-ime-multiple-messages

Fix: IME causes multiple messages on Enter (Fixes #2272)

HashWarlock committed:
fix: fix derive key to use path instead of subject for salt, update ra quote for verifiability

HashWarlock committed:
fix unit tests and update package json

shakkernerd committed:
Merge pull request #2303 from Phala-Network/fix/ra-and-derive-key

fix: derive key and update remote attestation

wtfsayo committed:
Merge branch 'develop' into direct-client_agent-crud

wtfsayo committed:
Merge pull request #2267 from jseparovic/direct-client_agent-crud

Improvement: Direct Client API - Add Delete Agent functionality

CryptoGraffe committed:
fix farcaster memory

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

shakkernerd committed:
chore: console -> elizaLogger

lalalune committed:
Update README.md

lalalune committed:
Merge pull request #2309 from elizaOS/lalalune-patch-2

Update README.md

shakkernerd committed:
Merge branch 'develop' into pr/CryptoGraffe/2307

wtfsayo committed:
Merge branch 'develop' into refactor-farcaster-client

wtfsayo committed:
Merge branch 'develop' into 1186-add-get-memories-by-ids

wtfsayo committed:
Merge pull request #2293 from bbopar/1186-add-get-memories-by-ids

feat: add getMemoryByIds to database adapters

wtfsayo committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update .env.example

wtfsayo committed:
Merge pull request #2281 from v1xingyue/character-from-url

feat: support load character from  character_url

shakkernerd committed:
Merge branch 'develop' into pr/augchan42/2248

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2248 from augchan42/feature/echochamber-conversation-starter

feat: (echochambers) add dead room detection and conversation starter

wtfsayo committed:
chore: Update README.md (#2280)

* Update README.md

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

0xnogo committed:
moving new locales

0xnogo committed:
Remove swipr from git tracking

ChainRex committed:
feat(plugin-sui): Add movepump action

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

Lukapetro committed:
fix: remove goku fallback image, env variable renamed

Lukapetro committed:
chore: resolve merge conflicts with develop

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/develop' into eternalai-agent

# Conflicts:
#	.env.example

ChainRex committed:
Merge branch 'develop' into rex-feat-plugin-sui

ChainRex commented:
POW
![pow](https://github.com/user-attachments/assets/d0f175c6-dc46-4c27-a274-2948cfa0b213)

Transaction:https://suiscan.xyz/mainnet/tx/3G7PqwkDtCtZNvwXSM4cb7RVJPuZvLYycotf8P4rQ8L6
Token:https://movepump.com/token/0xf94928f8da383439908b5d0ae32dfaf51ff74a95f9ef46f9d1b97b007f5d68b4::eliza::ELIZA

shakkernerd committed:
Merge pull request #2087 from sin-bufan/refactor-farcaster-client

refactor: farcaster client env configuration

ChainRex commented:
@coderabbitai review

Lukapetro committed:
chore: resolve merge conflicts with develop

ChainRex committed:
refactor(plugin-sui): rename buyAmountInToken

Commit:
Merge branch 'elizaOS:develop' into eternalai-agent

ChainRex committed:
Merge branch 'rex-feat-plugin-sui' of github.com:ChainRex/eliza into rex-feat-plugin-sui

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/eternalai-agent' into eternalai-agent

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

shakkernerd committed:
Merge pull request #2319 from eternalai-org/eternalai-agent

feat: Onchain Agent Transformer - transform any Eliza agents into unstoppable Solidity smart contracts deployed on 10+ blockchains

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

wtfsayo committed:
Merge branch 'develop' into pr/1865

wtfsayo committed:
Delete pnpm-lock.yaml

wtfsayo committed:
Create pnpm-lock.yaml

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Merge pull request #1865 from hellopleasures/develop

feat: add Dex Screener plugin with token price action, evaluators, an…

0xSero commented:
Is there a reason you're specifically using bytecode moves instead of a more readable form of code?

tcm390 committed:
Merge branch 'develop' into refactor-websearch

wtfsayo committed:
Merge pull request #2195 from chuasonglin1995/refactor-websearch

refactor: websearch into a service

everimbaq committed:
fix: correct OPENROUTER_API_KEY env in config

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

shakkernerd committed:
Merge pull request #2324 from xNomad-AI/main

fix: correct OPENROUTER_API_KEY env in config

leeran7 committed:
Feat: Enables multiple remote urls

leeran7 committed:
Chore: Remove unneeded spaces

leeran7 committed:
Merge branch 'develop' into multiple-remote-urls

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

sukrucildirr committed:
Update README_TR.md

derrekcoleman committed:
Fix formatting in Browser section of README.md

Hey all! I saw strange formatting in the README.md and figured it was an extra three backticks ```

Sure enough, deleting those made the section render as intended. I also reworded one of the sentences that had three "and"s to make it a little more legible.

JoeyKhd committed:
upd: added initial docker image fixes

JoeyKhd committed:
upd: we need all packages in Dockerfile workdir

rferrari committed:
feat: groq image vision provider

JoeyKhd committed:
upd: reworked docker ignore

JoeyKhd committed:
upd: rebuilt docs and bumped version numbers

JoeyKhd committed:
upd: resolved Dockerfile.docs build issues

JoeyKhd committed:
upd: reverted org generation in api docs

JoeyKhd committed:
upd: bumped types packages for docusaurus

thomasWos committed:
Update README.md

shakkernerd committed:
Merge branch 'develop' into pr/JoeyKhd/2347

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2347 from JoeyKhd/fix-dockerimage

fix: docker images+compose and broken documentation

mcp97 committed:
Merge branch 'develop' into patch-2

mcp97 committed:
Merge pull request #2346 from thomasWos/patch-2

Fix github client README.md

mcp97 committed:
Merge pull request #2342 from rferrari/feature/groq-image-vision-provider

feat: groq image vision provider

mcp97 committed:
Merge pull request #2220 from ai16z-demirix/fix/issue-2164

improvement: using strict types to avoid erorrs like issue 2164

augchan42 committed:
feat: (ragKnowledge) Enhance RAG knowledge handling

- Added support for double-byte characters (e.g., Chinese) to ensure proper processing.
- Implemented cleanup of deleted knowledge files to maintain data integrity.
- Enabled loading knowledge from directories, simplifying configuration and management.

These improvements address performance and usability issues, facilitating better support for large knowledge sets.

Closes #2323

augchan42 committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

tcm390 committed:
Merge pull request #2336 from derrekcoleman/patch-1

Fix formatting in Browser section of README.md

tcm390 committed:
Merge pull request #2334 from sukrucildirr/develop

Update README_TR.md

C0ldSmi1e committed:
fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

odilitime committed:
Merge pull request #2351 from augchan42/feature/rag-knowledge-improvements-v2

feat: (ragKnowledge) Enhance RAG knowledge handling

odilitime committed:
add back farcaster

HashWarlock committed:
fix: esure RA Action reply does not hallucinate

omahs committed:
Update README_FR.md

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

odilitime committed:
Merge pull request #1964 from Lukapetro/feature/add-client-instagram

feat: add instagram client

odilitime committed:
Merge pull request #2356 from omahs/patch-2

docs: Update README_FR.md

odilitime committed:
Merge pull request #2355 from elizaOS/HashWarlock-fix-ra-action-reply

fix: esure RA Action reply does not hallucinate

odilitime committed:
Merge pull request #2353 from C0ldSmi1e/develop

fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into ng/moving-readmes

proteanx committed:
feat: add safe_mode (& cfg_scale) for venice image generation (#2354)

* venice: add safe_mode & cfg_scale for images

* add missing types

for safeMode & cfgScale

* update lockfile

---------

Co-authored-by: Sayo <hi@sayo.wtf>

kesar committed:
Update package.json with proper dependencies (#2301)

these packages are being used in prod build

Co-authored-by: Sayo <hi@sayo.wtf>

odilitime committed:
Merge pull request #2149 from 0xnogo/ng/moving-readmes

chore: Reorganizing README translations into a dedicated i18n directory structure

odilitime committed:
bump version

odilitime committed:
bump lockfile

odilitime committed:
Merge pull request #2359 from elizaOS/odi-fix

chore: Prep 0.1.9-alpha.1

odilitime committed:
fix lint issues

odilitime committed:
Merge pull request #2328 from leeran7/multiple-remote-urls

feat: Enable Multiple remote character urls

odilitime committed:
Merge pull request #2368 from elizaOS/odi-fix

chore: lint

defitricks committed:
fix: Fix typo in import statement for Solana AgentKit plugin Update i… (#2370)

* fix: Fix typo in import statement for Solana AgentKit plugin Update index.ts

* Update index.ts

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

0xCardiE committed:
feat: dexscreener trending (#2325)

* add to main index

* cleanup

---------

Co-authored-by: Sayo <hi@sayo.wtf>

tcm390 committed:
check if remote url is defined first (#2384)

0xSero committed:
Startup Error: ENV misconfig (#2378)

* fix

* Update info.json

---------

Co-authored-by: Sero <69639595+Seroxdesign@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

dev-whoan committed:
fix: check env REMOTE_CHARACTER_URLS is defined (#2382)

Co-authored-by: Sayo <hi@sayo.wtf>

ericlehong committed:
Fix typo (#2385)

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

---------

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

wtfsayo committed:
Merge branch 'develop' into rex-feat-plugin-sui

wtfsayo commented:
@ChainRex where did you get the bytecode from? can you do remote fetch?

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.
